pkg/discovery/queries.go

```diff
@@ -269,15 +274,21 @@ func QueryPodLogs(kubeClient kubernetes.Interface, recorder *QueryRecorder, ns s
 		}
 	}
 
-	visitedPods := make(map[string]bool)
-	logrus.Infof("Collecting Pod Logs (%v)", ns)
-
-	for _, filedSelector := range cfg.Limits.PodLogs.FieldSelectors {
-		opts.FieldSelector = filedSelector
+	if len(ns) > 0 {
+		logrus.Infof("Collecting Pod Logs by namespace (%v)", ns)
 		err := gatherPodLogs(kubeClient, ns, opts, cfg, visitedPods)
 		if err != nil {
 			return err
 		}
+	} else {
+		logrus.Infoln("Collecting Pod Logs by FieldSelectors", cfg.Limits.PodLogs.FieldSelectors)
+		for _, filedSelector := range cfg.Limits.PodLogs.FieldSelectors {
```

Comment:
s/filedSelector/fieledSelector or maybe even s/filedSelector/fs since it is such a small context.